### ISSUE TITLE
Implement a better solution for checkpointing in pycbc inference

### ISSUE THREAD

**Issue by duncan-brown** [ligo-cbc/pycbc#2078]
We need a better solution than the one in https://github.com/ligo-cbc/pycbc/pull/2055 for checkpointing so that Condor and Pegasus can manage workflows with PE jobs that checkpoint.

**Comment by cdcapano** [comment]
@duncan-brown Could you give some more details on what you tried, and what didn't work?

**Comment by cdcapano** [comment]
No, checkpointing is still done in the same way. Maybe the pre-thinning to make files smaller is implemented we can revisit this.
